src/Atmos/Model/AtmosModel.jl

```diff
@@ -0,0 +1,134 @@
+  init_aux!, init_state!, update_aux!
+  
+struct AtmosModel{T,M,R,S,BC,IS} <: BalanceLaw
+  turbulence::T
+  moisture::M
+  radiation::R
+  # TODO: a better mechanism than functions.
+  source::S
+  boundarycondition::BC
+  init_state::IS
+end
+
+function vars_state(m::AtmosModel, T)
+  NamedTuple{(:ρ, :ρu, :ρe, :turbulence, :moisture, :radiation), 
+  Tuple{T, SVector{3,T}, T, vars_state(m.turbulence,T), vars_state(m.moisture,T), vars_state(m.radiation, T)}}
+end
+function vars_transform(m::AtmosModel, T)
+  NamedTuple{(:u, :turbulence, :moisture, :radiation),
+  Tuple{SVector{3,T}, vars_transform(m.turbulence,T), vars_transform(m.moisture,T), vars_transform(m.radiation,T)}}
+end
+function vars_diffusive(m::AtmosModel, T)
+  NamedTuple{(:ρτ, :turbulence, :moisture, :radiation),
+  Tuple{SVector{6,T}, vars_diffusive(m.turbulence,T), vars_diffusive(m.moisture,T), vars_diffusive(m.radiation,T)}}
+end
+function vars_aux(m::AtmosModel, T)
+  NamedTuple{(:coord, :turbulence, :moisture, :radiation),
+  Tuple{NamedTuple{(:x,:y,:z),Tuple{T,T,T}}, vars_aux(m.turbulence,T), vars_aux(m.moisture,T), vars_aux(m.radiation,T)}}
+end
+
+# Navier-Stokes flux terms
+function flux!(m::AtmosModel,flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  # preflux
+  ρinv = 1/state.ρ
+  ρu = state.ρu
+  u = ρinv * ρu
+  
+  p = pressure(m.moisture, state, aux, t)
+
+  # invisc terms
+  flux.ρ  = ρu 
+  flux.ρu = ρu .* u' + p*I
+  flux.ρe = u * (state.ρe + p)
+
+  flux_diffusive!(m, flux, state, diffusive, aux, t)
+end
+
+function flux_diffusive!(m::AtmosModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  u = (1/state.ρ) * state.ρu
+
+  # diffusive
+  ρτ11, ρτ22, ρτ33, ρτ12, ρτ13, ρτ23 = diffusive.ρτ
+  ρτ = SMatrix{3,3}(ρτ11, ρτ12, ρτ13,
+                    ρτ12, ρτ22, ρτ23,
+                    ρτ13, ρτ23, ρτ33)
+  flux.ρu += ρτ
+  flux.ρe += ρτ*u
+
+  # moisture-based diffusive fluxes
+  flux_diffusive!(m.moisture, flux, state, diffusive, aux, t)
+end
+
+function wavespeed(m::AtmosModel, nM, state::Vars, aux::Vars, t::Real)
+  ρinv = 1/state.ρ
+  ρu = state.ρu
+  u = ρinv * ρu
+  return abs(dot(nM, u)) + soundspeed(m.moisture, state, aux, t)
+end
+
+function gradtransform!(m::AtmosModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+  ρinv = 1 / state.ρ
+  transform.u = ρinv * state.ρu
+
+  gradtransform!(m.moisture, transform, state, aux, t)
+end
+
+function diffusive!(m::AtmosModel, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real)
+  ∇u = ∇transform.u
+  
+  # strain rate tensor
+  # TODO: we use an SVector for this, but should define a "SymmetricSMatrix"? 
+  S = SVector(∇u[1,1],
+              ∇u[2,2],
+              ∇u[3,3],
+              (∇u[1,2] + ∇u[2,1])/2,
+              (∇u[1,3] + ∇u[3,1])/2,
+              (∇u[2,3] + ∇u[3,2])/2)
+
+  # kinematic viscosity tensor
+  ρν = scaled_kinematic_viscosity_tensor(m.turbulence, S, state, aux, t)
```

Comment:
Maybe use `dynamic_viscosity_tensor` in place of `scaled_kinematic_viscosity_tensor` (because `\rho \nu = \mu = dynamic viscosity')?

Reply:
any suggestions for "scaled momentum flux tensor"?

Reply:
I was wondering about it. The oddity is that what you call scaled momentum flux strictly is the momentum flux (per unit volume), whereas what is called the momentum flux (here and generally in the community) is the specific momentum flux (per unit mass). If you want to be more explicit for `scaled`, a good nomenclature would be something like `momentum_flux_by_volume` (but it gets long). `Scaled` is acceptable.